sdk-design-doc-template.md

```diff
@@ -0,0 +1,83 @@
+# SDK Design Doc Template
+
+Template for SDK design docs that will be checked into a SDK design docs repo.
+
+## Summary
+
+Brief description of the design, what it’s meant to accomplish, how it should be used, etc. Any information that would be relevant to the usage of the new design should be provided. Any changes to previous workflows that may result from the new design should be outlined. Links to relevant HIPs, protobuf PRs, services PRs, etc. should be provided here as well to provide additional context for the use of the new design. Links to relevant SDK/TCK issues/PRs should be provided as well.
+
+## New APIs
+
+Define the names of the new APIs being added, as well as how they are interacted with.
+
+### New API #1
+
+- `Type property` : description of the property.
```

Comment:
The definition of the API can not be language specific in this doc. Based on that we need to define some pattern how the language-agnostic API definition should look like.